worker/src/backgroundMigrations/migrateObservationsFromPostgresToClickhouse.ts

```diff
@@ -0,0 +1,160 @@
+import { IBackgroundMigration } from "./IBackgroundMigration";
+import { clickhouseClient, logger } from "@langfuse/shared/src/server";
+import { parseArgs } from "node:util";
+import { prisma, Prisma } from "@langfuse/shared/src/db";
+import { env } from "../env";
+import { Observation } from "@prisma/client";
+
+async function addTemporaryColumnIfNotExists() {
+  const columnExists = await prisma.$queryRaw<{ column_exists: boolean }[]>(
+    Prisma.sql`
+      SELECT EXISTS (
+        SELECT 1
+        FROM information_schema.columns
+        WHERE table_name = 'observations'
+        AND column_name = 'tmp_migrated_to_clickhouse'
+      ) AS column_exists;
+    `,
+  );
+  if (!columnExists[0]?.column_exists) {
+    await prisma.$executeRaw`ALTER TABLE observations ADD COLUMN tmp_migrated_to_clickhouse BOOLEAN DEFAULT FALSE;`;
+    logger.info("Added temporary column tmp_migrated_to_clickhouse");
+  } else {
+    logger.info(
+      "Temporary column tmp_migrated_to_clickhouse already exists. Continuing...",
+    );
+  }
+}
+
+export default class MigrateObservationsFromPostgresToClickhouse
+  implements IBackgroundMigration
+{
+  private isAborted = false;
+
+  async validate(
+    args: Record<string, unknown>,
+  ): Promise<{ valid: boolean; invalidReason: string | undefined }> {
+    if (!env.CLICKHOUSE_URL) {
+      return {
+        valid: false,
+        invalidReason: "Clickhouse URL must be configured to perform migration",
+      };
+    }
+    return { valid: true, invalidReason: undefined };
+  }
+
+  async run(args: Record<string, unknown>): Promise<void> {
+    logger.info(
+      `Migrating observations from postgres to clickhouse with ${JSON.stringify(args)}`,
+    );
+
+    const maxRowsToProcess = Number(args.maxRowsToProcess ?? Infinity);
+    const batchSize = Number(args.batchSize ?? 5000);
+    const maxDate = new Date((args.maxDate as string) ?? new Date());
+
+    await addTemporaryColumnIfNotExists();
+
+    let processedRows = 0;
+    while (!this.isAborted && processedRows < maxRowsToProcess) {
+      const observations = await prisma.$queryRaw<
+        Array<Observation>
+      >(Prisma.sql`
+        SELECT id, trace_id, project_id, type, parent_observation_id, start_time, end_time, name, metadata, level, status_message, version, input, output, unit, prompt_id, input_cost, output_cost, total_cost, created_at, updated_at
```

Comment:
@maxdeichmann I find it hard to map the following fields: `provided_model_name, model_parameters, provided_input_usage_units, provided_output_usage_units, provided_total_usage_units, input_usage_units, output_usage_units, total_usage_units, provided_input_cost, provided_output_cost, provided_total_cost, prompt_name, prompt_version`. Could you provide more context on what they mean and how we want to transition existing data to them?

Reply:
Ah yes, see my other comment below with the commit. Happy to jump on a call to discuss.